Examples/MAX78000/I2S_DMA_Target/main.c

```diff
@@ -0,0 +1,224 @@
+ * property whatsoever. Maxim Integrated Products, Inc. retains all
+ * ownership rights.
+ *
+ ******************************************************************************/
+
+/**
+ * @file    main.c
+ * @brief   Main for I2S CODEC loopback
+ * @details I2S CODEC loopback using DMA
+ */
+
+#include <stdio.h>
+#include <stdint.h>
+#include "mxc.h"
+#include "max9867.h"
+
+#undef USE_I2S_INTERRUPTS
+
+#define CODEC_I2C MXC_I2C0
+#define CODEC_I2C_FREQ 100000
+
+#define CODEC_MCLOCK 12288000
+#define BITS_PER_CHANNEL 16
+#define CHANNELS_PER_FRAME 2
+#define SAMPLE_RATE 24000
+
+#define BIT_CLK (SAMPLE_RATE * CHANNELS_PER_FRAME * BITS_PER_CHANNEL)
+#define CLK_DIV (((CODEC_MCLOCK / 2) / BIT_CLK) - 1)
+
+#define I2S_DMA_BUFFER_SIZE 64
+
+volatile int dma_flag;
+uint32_t i2s_rx_buffer[I2S_DMA_BUFFER_SIZE * 2];
+int dma_ch_tx, dma_ch_rx;
+uint32_t *rxBufPtr = i2s_rx_buffer;
+
+void blink_halt(const char *msg)
+{
+    if (msg && *msg)
+        puts(msg);
+
+    for (;;) {
+        LED_On(LED1);
+        LED_Off(LED2);
+        MXC_Delay(MXC_DELAY_MSEC(500));
+        LED_On(LED2);
+        LED_Off(LED1);
+        MXC_Delay(MXC_DELAY_MSEC(500));
+    }
+}
+
+void dma_handler(void)
+{
+    dma_flag = 1;
+    MXC_DMA_Handler();
+}
+
+void dma_init(void)
+{
+    MXC_NVIC_SetVector(DMA0_IRQn, dma_handler);
+    MXC_NVIC_SetVector(DMA1_IRQn, dma_handler);
+    NVIC_EnableIRQ(DMA0_IRQn);
+    NVIC_EnableIRQ(DMA1_IRQn);
+}
+
+void dma_callback(int channel, int result)
+{
+    static uint32_t *tx_buf = i2s_rx_buffer + I2S_DMA_BUFFER_SIZE;
+
+    if (channel == dma_ch_tx) {
+        MXC_DMA_ReleaseChannel(dma_ch_tx);
+        dma_ch_tx = MXC_I2S_TXDMAConfig(tx_buf, I2S_DMA_BUFFER_SIZE * sizeof(i2s_rx_buffer[0]));
+
+    } else if (channel == dma_ch_rx) {
+        tx_buf = rxBufPtr;
+
+        if (rxBufPtr == i2s_rx_buffer) {
+            rxBufPtr = i2s_rx_buffer + I2S_DMA_BUFFER_SIZE;
+        } else {
+            rxBufPtr = i2s_rx_buffer;
+        }
+        MXC_DMA_ReleaseChannel(dma_ch_rx);
+        dma_ch_rx = MXC_I2S_RXDMAConfig(rxBufPtr, I2S_DMA_BUFFER_SIZE * sizeof(i2s_rx_buffer[0]));
+    }
+}
+
+void dma_work_loop(void)
+{
+    int trig = 0;
+
+    dma_init();
+    MXC_I2S_RegisterDMACallback(dma_callback);
+    dma_ch_tx = MXC_I2S_TXDMAConfig(i2s_rx_buffer + I2S_DMA_BUFFER_SIZE,
+                                    I2S_DMA_BUFFER_SIZE * sizeof(i2s_rx_buffer[0]));
+    dma_ch_rx = MXC_I2S_RXDMAConfig(i2s_rx_buffer, I2S_DMA_BUFFER_SIZE * sizeof(i2s_rx_buffer[0]));
+
+    for (;;) {
+        if (dma_flag) {
+            dma_flag = 0;
+            /*
+        dma activity triggered work
+      */
+            if (++trig == SAMPLE_RATE / I2S_DMA_BUFFER_SIZE) {
+                trig = 0;
+                LED_Toggle(LED2);
+            }
+        }
+        /*
+      non-dma activity triggered work
+    */
+    }
+}
+
+void i2c_init(void)
+{
+    if (MXC_I2C_Init(CODEC_I2C, 1, 0))
+        blink_halt("Error initializing I2C controller");
```

Comment:
Explicitly comparing the return value against E_NO_ERROR would be better.

Reply:
Fixed in https://github.com/Analog-Devices-MSDK/msdk/pull/240/commits/1c5ae59594fba5650b3ca48105437a28f0cc9f0f